subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesBetweenBuildsFileSystemWatchingIntegrationTest.groovy

```diff
@@ -49,6 +49,53 @@ class ChangesBetweenBuildsFileSystemWatchingIntegrationTest extends AbstractFile
         executedAndNotSkipped ":compileJava", ":classes", ":run"
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/19353")
+    def "source file changes are recognized when file length does not change"() {
+        buildFile << """
+            apply plugin: "application"
```

Comment:
I think the test should be much simpler: Let's define a custom task with an input and an output and use that instead of the big `application` plugin.

Reply:
Ok, I will change it and move it to `UpToDateIntegTest`